library/src/main/java/com/dslplatform/json/ExternalConverterAnalyzer.java

```diff
@@ -0,0 +1,78 @@
+package com.dslplatform.json;
+
+import java.lang.reflect.Type;
+import java.util.*;
+
+class ExternalConverterAnalyzer {
+    private final Set<String> lookedUpClasses = new HashSet<String>();
+
+    final DslJson.ConverterFactory<JsonWriter.WriteObject> writerFactory = new DslJson.ConverterFactory<JsonWriter.WriteObject>() {
+        @Nullable
+        @Override
+        public JsonWriter.WriteObject<?> tryCreate(Type manifest, DslJson dslJson) {
+            tryFindAndRegisterExternalConverter(manifest, dslJson);
+            return dslJson.getRegisteredEncoder(manifest);
+        }
+    };
+
+    final DslJson.ConverterFactory<JsonReader.ReadObject> readerFactory = new DslJson.ConverterFactory<JsonReader.ReadObject>() {
+        @Nullable
+        @Override
+        public JsonReader.ReadObject<?> tryCreate(Type manifest, DslJson dslJson) {
+            tryFindAndRegisterExternalConverter(manifest, dslJson);
+            return dslJson.getRegisteredDecoder(manifest);
+        }
+    };
+
+    final DslJson.ConverterFactory<JsonReader.BindObject> binderFactory = new DslJson.ConverterFactory<JsonReader.BindObject>() {
+        @Nullable
+        @Override
+        public JsonReader.BindObject<?> tryCreate(Type manifest, DslJson dslJson) {
+            tryFindAndRegisterExternalConverter(manifest, dslJson);
+            return dslJson.getRegisteredBinder(manifest);
+        }
+    };
+
+    private synchronized void tryFindAndRegisterExternalConverter(final Type manifest, DslJson<?> dslJson) {
+        if (!(manifest instanceof Class<?>)) {
+            return;
+        }
+        List<String> converterClassNames = resolveExternalConverterClassNames(((Class<?>) manifest).getName());
+        if (converterClassNames == null) {
+            return;
+        }
+        for (ClassLoader cl : dslJson.classLoaders) {
```

Comment:
Class loaders seems to be used only here

Reply:
Moved to analyzer class